nexus/src/db/error.rs

```diff
@@ -88,21 +74,44 @@ pub fn public_error_from_diesel_pool_create(
     resource_type: ResourceType,
     object_name: &str,
 ) -> PublicError {
+    public_error_from_diesel_pool_helper(error, |error| {
+        public_error_from_diesel_create(error, resource_type, object_name)
+    })
+}
+
+/// Converts a Diesel pool error to an external error for a case where we do not
+/// expect the query to fail and the error cannot be easily summarized for an
+/// end user
+pub fn public_error_from_diesel_pool_shouldnt_fail(
```

Comment:
@smklein FYI I refactored the functions here a bit so I could handle things a bit differently during Nexus startup.